File: api/Models/Recipe.cs
namespace api.Models
{
    public class Recipe
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public int RecipeCategoryId { get; set; } = 0;
        public RecipeCategory? RecipeCategory { get; set; }
        public string Steps { get; set; } = string.Empty;
        public string? ImageUrl { get; set; } = string.Empty;
        public bool IsPublished { get; set; } = false;
        public int UserId { get; set; } = 0;
        public User? User { get; set; }
        public ICollection<ProductInRecipe>? ProductsInRecipes { get; set; } = null!;
        public ICollection<RecipeReview>? Reviews { get; set; } = null!;
    }
}

File: webApp/src/app/shared/models/recipe.ts
import { User } from "./user";

export class Recipe {
  id: number = 0;
  title: string = '';
  description: string = '';
  recipeCategoryId: number = 0;
  steps: string = '';
  imageUrl: string = '';
  isPublished: number = 0;
  categoryId: number = 0;
  userId: number = 0;
  user: User = new User();
  image: string = '';
}
